### PR TITLE
Bump to v3.0.0 (remove serving code and update README)

### PR DESCRIPTION
**Pull request by laurenyu**
*Description of changes:*
Starting with MXNet version 1.4.0, training and serving have different images. Now that https://github.com/aws/sagemaker-mxnet-serving-container is public, there's no need to keep the serving code/tests in this repository.

I've also removed the MXNet 1.3.0 Dockerfiles, following the pattern set with 1.3.0, given that those Dockerfiles are no longer compatible with the current code in this repository (since they expect serving code).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://sbpy3lu9z2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=16abbc9d-68e2-4afd-85ce-2ec3b88c677d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: README.rst [hunk 1]
```diff
@@ -65,11 +68,17 @@ To create the SageMaker MXNet Container Python package:
     cp dist/sagemaker_mxnet_container-<package_version>.tar.gz docker/<mxnet_version>/final
 
 Once you have those binaries, you can then build the image.
-The Dockerfiles expect two build arguments:
 
-- ``py_version``: the Python version.
+
+For MXNet versions 1.1.0-1.3.0, the Dockerfiles expect two build arguments:
```
Comment (ChoiByungWook) [inline README.rst]: hmmm, do you think version specifics should exist in their respective branch/tag?

File: README.rst [hunk 2]
```diff
@@ -95,12 +104,10 @@ Don't forget the period at the end of the command!
     # Example
 
     # CPU
-    docker build -t preprod-mxnet:1.1.0-cpu-py3 --build-arg py_version=3 \
-    --build-arg framework_installable=mxnet-1.1.0-py2.py3-none-manylinux1_x86_64.whl -f Dockerfile.cpu .
+    docker build -t preprod-mxnet:1.1.0-cpu-py3 --build-arg py_version=3.6 -f Dockerfile.cpu .
```
Comment (ChoiByungWook) [inline README.rst]: For training py_version should be just '3'

Comment (laurenyu) [reply README.rst]: ah, my bad. didn't realize it was just serving.

File: README.rst (hunk 1 above)
Comment (laurenyu) [reply README.rst]: that could clean the README up quite a bit. how about I do it in a following PR?

File: README.rst (hunk 2 above)
Comment (ChoiByungWook) [reply README.rst]: no problem, it is kind of a random change.

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://sbpy3lu9z2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=857871c7-0014-4269-bbab-d6b995529d16%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://sbpy3lu9z2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=c62d9df7-bc08-41c6-b494-197fb19ba39f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://sbpy3lu9z2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=7d92cbbd-120f-4d06-ac1e-1f70310d4920%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: FAILED
* [Build Logs](https://sbpy3lu9z2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=b9d90a60-d18d-46f6-87f4-d1294acec5ef%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://sbpy3lu9z2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=dfbe9c63-81c2-445d-a8b1-4a109a72cf11%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: buildspec.yml [hunk 3]
```diff
@@ -32,39 +30,24 @@ phases:
       # run unit tests
       - pytest test/unit
 
-      # Create pip archive
+      # Create pip archive in dist/
+      # tar_name will be something like: sagemaker_mxnet_container-3.0.0.tar.gz
+      - python3 setup.py sdist
```
Comment (ChoiByungWook) [inline buildspec.yml]: does python3 vs python2.7 affect this?

File: docker/1.4.0/final/Dockerfile.cpu [hunk 4]
```diff
@@ -22,13 +22,13 @@ RUN cd /tmp && \
 
 WORKDIR /
 
-COPY dist/sagemaker_mxnet_container-2.0.0.tar.gz /sagemaker_mxnet_container-2.0.0.tar.gz
+COPY dist/sagemaker_mxnet_container-3.0.0.tar.gz /sagemaker_mxnet_container-3.0.0.tar.gz
 
 RUN pip install --no-cache mxnet-mkl==1.4.0.post0 \
```
Comment (ChoiByungWook) [inline docker/1.4.0/final/Dockerfile.cpu]: apologies, this isn't in your PR scope, but I think this should be pinned at 1.4.0 and not post as stated here: https://github.com/aws/sagemaker-mxnet-container/pull/79#discussion_r281397147

File: docker/1.4.0/final/Dockerfile.gpu [hunk 5]
```diff
@@ -22,13 +22,13 @@ RUN cd /tmp && \
 
 WORKDIR /
 
-COPY dist/sagemaker_mxnet_container-2.0.0.tar.gz /sagemaker_mxnet_container-2.0.0.tar.gz
+COPY dist/sagemaker_mxnet_container-3.0.0.tar.gz /sagemaker_mxnet_container-3.0.0.tar.gz
 
 RUN pip install --no-cache mxnet-cu92mkl==1.4.0.post0 \
```
Comment (ChoiByungWook) [inline docker/1.4.0/final/Dockerfile.gpu]: same thing here

File: test/integration/local/test_mnist_training.py [hunk 6]
```diff
@@ -28,40 +27,20 @@
...
-            prediction = predictor.predict(data)
-        finally:
-            mx.delete_endpoint()
 
-    # Check that there is a probability for each possible class in the prediction
-    prediction_values = prediction.decode('utf-8').split(',')
-    assert len(prediction_values) == 10
+def test_distributed(docker_image, sagemaker_local_session, framework_version, processor, tmpdir):
+    if processor == 'gpu':
+        pytest.skip('Local Mode does not support distributed training on GPU.')
```
Comment (ChoiByungWook) [inline test/integration/local/test_mnist_training.py]: does this fail when using gpu?

File: buildspec.yml (hunk 3 above)
Comment (laurenyu) [reply buildspec.yml]: not sure, but we've always done this. a better approach might be to use bdist_wheel like Chainer/PyTorch do, which generates a binary that's explicitly for both Python versions, but I think that's out of scope here.

File: docker/1.4.0/final/Dockerfile.cpu (hunk 4 above)
Comment (laurenyu) [reply docker/1.4.0/final/Dockerfile.cpu]: should I run a git blame... 😂 

(fixed)

File: docker/1.4.0/final/Dockerfile.gpu (hunk 5 above)
Comment (laurenyu) [reply docker/1.4.0/final/Dockerfile.gpu]: fixed

File: test/integration/local/test_mnist_training.py (hunk 6 above)
Comment (laurenyu) [reply test/integration/local/test_mnist_training.py]: yeah. see previous builds on this PR (also ran into this last week when doing local runs)

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://sbpy3lu9z2.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=45b354a8-71d6-4b3f-b885-7d557183374f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**